src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -326,6 +331,23 @@ def _slice(
         assert common.is_relative_index_sequence(slice_)
         return new_domain, slice_
 
+    if dace:
+        # Extension of NdArrayField : Support for SDFGConvertible GT4Py Programs
+        def data_ptr(self) -> int:
+            obj = self.ndarray
+            if dace.dtypes.is_array(obj) and (
+                hasattr(obj, "__array_interface__") or hasattr(obj, "__cuda_array_interface__")
+            ):
+                if dace.dtypes.is_gpu_array(obj):
+                    return obj.__cuda_array_interface__["data"][0]  # type: ignore
+                else:
+                    return self.ndarray.__array_interface__["data"][0]  # type: ignore
+            else:
+                raise ValueError("Unsupported data container.")
```

Comment:
This should work for both numpy and cupy arrays. In the future we should try to only dlpack, but we need to write a couple of utility functions with ctypes to access the dlpack data structures so for now we're also using this trick in the storage buffer allocators.
```suggestion
            return self.array_ns.byte_bounds(self.ndarray)[0]
```